openapi-core/src/main/java/io/ballerina/openapi/core/generators/client/mock/MockClientFunctionGenerator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.openapi.core.generators.client.mock;
```

Comment:
Why are we having this in a separate package, where as the other mock classes are in the cores/generators/client

Reply:
Grouped this into separate packages to enhance the readability since these classes include the implementation for one purpose.

Reply:
Then isn't it better to have the mock module inside the client package and move all the mock related class to that. I think we need to move the status code binding related classes also to a separate module for readability.

Reply:
yeah let's move other status-binding related classes also into separate package via separate PR